### DISCUSSION TITLE
zookeeper fail to start due to unknown jmx configuration null

### DISCUSSION THREAD

**Discussion by gaobosince1987** [strimzi/strimzi-kafka-operator#9309] [General]
I tried to use build 0.38 strimzi with 0.36 kafka, kafka server works fine, but zookeeper fail to start, error indicate jmx configuration is null, I wonder is there some configuration I should add for jmx? Not sure what jmx configuration zookeeper is complaining here.

Strimzi helm config:
  &emsp;      pod:
  &emsp;&emsp;        metadata:
   &emsp;&emsp;&emsp;         labels:
  &emsp;&emsp;&emsp;&emsp;            app: {{ .Values.kafka.clusterName }}
  &emsp;          annotations:
  &emsp;&emsp;            ad.datadoghq.com/kafka.check_names: '["kafka-jmx"]'
 &emsp;&emsp;             ad.datadoghq.com/kafka.init_configs: '[{"is_jmx": true, "collect_default_metrics": true}]'
  &emsp;&emsp;            ad.datadoghq.com/kafka.instances: '[{"host": "%%host%%","port":9999,"max_returned_metrics":20000}]'
              
Zookeeper log:
exec /usr/bin/tini -w -e 143 -- /opt/kafka/bin/zookeeper-server-start.sh /tmp/zookeeper.properties
Exception in thread "main" java.lang.reflect.InvocationTargetException
 &emsp;   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 &emsp;   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
 &emsp;   at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 &emsp;   at java.base/java.lang.reflect.Method.invoke(Method.java:568)
 &emsp;   at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:491)
  &emsp;  at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:503)
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.length()" because "this.text" is null
 &emsp;   at java.base/java.util.regex.Matcher.getTextLength(Matcher.java:1769)
 &emsp;   at java.base/java.util.regex.Matcher.reset(Matcher.java:415)
 &emsp;   at java.base/java.util.regex.Matcher.<init>(Matcher.java:252)
 &emsp;   at java.base/java.util.regex.Pattern.matcher(Pattern.java:1134)
 &emsp;   at io.prometheus.jmx.JavaAgent.parseConfig(JavaAgent.java:52)
 &emsp;   at io.prometheus.jmx.JavaAgent.premain(JavaAgent.java:26)
    ... 6 more
*** java.lang.instrument ASSERTION FAILED ***: "!errorOutstanding" with message Outstanding error when calling method in invokeJavaAgentMainMethod at src/java.instrument/share/native/libinstrument/JPLISAgent.c line: 619
*** java.lang.instrument ASSERTION FAILED ***: "success" with message invokeJavaAgentMainMethod failed at src/java.instrument/share/native/libinstrument/JPLISAgent.c line: 459
*** java.lang.instrument ASSERTION FAILED ***: "result" with message agent load/premain call failed at src/java.instrument/share/native/libinstrument/JPLISAgent.c line: 422
FATAL ERROR in native method: processing of -javaagent failed, processJavaStart failed
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
V  [libjvm.so+0x8cea25]  jni_FatalError+0xc5
V  [libjvm.so+0xa343ad]  JvmtiExport::post_vm_initialized()+0x14d
V  [libjvm.so+0xec8417]  Threads::create_vm(JavaVMInitArgs*, bool*)+0x847
V  [libjvm.so+0x8e9ec5]  JNI_CreateJavaVM+0x65
C  [libjli.so+0x8f7d]  JavaMain+0x9d
C  [libjli.so+0xd3dd]  ThreadJavaMain+0xd

**Comment by scholzj** [comment]
Please format the code parts to make them readable. You should also share your configuration and explain what exactly are you trying to do.

**Comment by gaobosince1987** [comment]
turns out to be zookeeper_run.sh have script referring to 0.20.0 instead of 0.18.0 java agent jar, once hard coded, the issue is fixed. 

  KAFKA_OPTS="$KAFKA_OPTS -javaagent:$(ls "$KAFKA_HOME"/libs/jmx_prometheus_javaagent*.jar)=9404:$KAFKA_HOME/custom-config/metrics-config.json"
